provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/controller/endpoint/EndpointController.java

```diff
@@ -1,3 +1,165 @@
 package gov.va.api.health.providerdirectory.service.controller.endpoint;
 
-public class EndpointController {}
+import gov.va.api.health.providerdirectory.service.AddressResponse;
+import gov.va.api.health.providerdirectory.service.CountParameter;
+import gov.va.api.health.providerdirectory.service.client.VlerClient;
+import gov.va.api.health.providerdirectory.service.controller.Bundler;
+import gov.va.api.health.providerdirectory.service.controller.Bundler.BundleContext;
+import gov.va.api.health.providerdirectory.service.controller.PageLinks.LinkConfig;
+import gov.va.api.health.providerdirectory.service.controller.Parameters;
+import gov.va.api.health.providerdirectory.service.controller.Validator;
+import gov.va.api.health.stu3.api.resources.Endpoint;
+import gov.va.api.health.stu3.api.resources.OperationOutcome;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import javax.validation.constraints.Min;
+import lombok.AllArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Request Mappings for Endpoint Resource, see
+ * http://www.fhir.org/guides/argonaut/pd/StructureDefinition-argo-endpoint.html for implementation
+ * details.
+ */
+@RestController
+@RequestMapping(
+  value = {"/api/Endpoint"},
+  produces = {"application/json", "application/fhir+json", "application/json+fhir"}
+)
+@AllArgsConstructor(onConstructor = @__({@Autowired}))
+public class EndpointController {
+
+  private Transformer transformer;
+
+  private Bundler bundler;
+
+  private VlerClient vlerClient;
+
+  private Endpoint.Bundle bundle(MultiValueMap<String, String> parameters, int page, int count) {
+    Pair<List<EndpointWrapper>, Integer> root = search(parameters);
+    LinkConfig linkConfig =
+        LinkConfig.builder()
+            .path("Endpoint")
+            .queryParams(parameters)
+            .page(page)
+            .recordsPerPage(count)
+            .totalRecords(root.getRight())
+            .build();
+    return bundler.bundle(
+        BundleContext.of(
+            linkConfig, root.getLeft(), transformer, Endpoint.Entry::new, Endpoint.Bundle::new));
+  }
+
+  /** Read by identifier. */
+  @GetMapping(value = {"/{publicId}"})
+  public Endpoint readByIdentifier(@PathVariable("publicId") String publicId) {
```

Comment:
In data-query you see `publicId` because there is translation between public-facing IDs and CDW IDs. Since we're not doing any ID translation, this can just be 'id'.